src/context.ml

```diff
@@ -251,6 +251,30 @@ let create ~(kind : Kind.t) ~path ~base_env ~env_extra ~name ~merlin ~use_findli
   let get_path var = Path.absolute (get var) in
   let stdlib_dir = get_path "standard_library" in
   let natdynlink_supported = Path.exists (Path.relative stdlib_dir "dynlink.cmxa") in
+  let env =
+    (* We redirect the output of all commands, so by default the compiler will disable
+       colors. Since we support colors in the output of commands, we force it via OCAMLPARAM
+       if stderr supports colors. *)
+    if Lazy.force Ansi_color.stderr_supports_colors then
+      let vars =
+        if get "version" <> "4.02.3" then
+          let value =
+            match get_env env "OCAMLPARAM" with
+            | None -> "color=always,_"
+            | Some s -> "color=always," ^ s
+          in
+          String_map.singleton "OCAMLPARAM" value
+        else
+          String_map.empty
+      in
+      let vars = match get_env env "OPAMCOLOR" with
+      | None -> String_map.add ~key:"OPAMCOLOR" ~data:"always" vars
+      | Some _ -> vars
+      in
+      extend_env ~env ~vars
```

Comment:
The binding `OCAMLPARAM=...` should go into the map `env_extra` as well. `env_extra` is already included in this `env`, but we print it as part of the s-expression for the build context in the log